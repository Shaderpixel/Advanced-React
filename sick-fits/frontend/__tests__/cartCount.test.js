import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import CartCountComponent from '../components/CartCount';

describe('<CartCountComponent />', () => {
  it('render ', () => {
    shallow(<CartCountComponent count={10} />);
  });

  it('match the snapshot', () => {
    const wrapper = shallow(<CartCountComponent count={11} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('update via props', () => {
    const wrapper = shallow(<CartCountComponent count={50} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
    wrapper.setProps({ count: 10 });
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should show mounting', () => {
    const wrapper = mount(<CartCountComponent count={50} />);
    console.log(wrapper.debug());
  });
});
