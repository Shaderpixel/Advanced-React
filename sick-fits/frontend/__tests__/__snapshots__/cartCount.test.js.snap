// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCountComponent /> match the snapshot 1`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="11"
      timeout={
        Object {
          "enter": 4000,
          "exit": 4000,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        11
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;

exports[`<CartCountComponent /> update via props 1`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="50"
      timeout={
        Object {
          "enter": 4000,
          "exit": 4000,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        50
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;

exports[`<CartCountComponent /> update via props 2`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="10"
      timeout={
        Object {
          "enter": 4000,
          "exit": 4000,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        10
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;
